be/src/exprs/math_functions.cpp

```diff
@@ -715,27 +708,32 @@ Status MathFunctions::rand_prepare(FunctionContext* context, FunctionContext::Fu
             }
 
             int64_t seed_value = ColumnHelper::get_const_value<TYPE_BIGINT>(seed_column);
-            seed = seed_value;
+            generator->seed(seed_value);
         } else {
-            seed = GetCurrentTimeNanos();
+            generator->seed(std::random_device()());
         }
-        context->set_function_state(scope, reinterpret_cast<void*>(seed)); // NOLINT
+        context->set_function_state(scope, generator); // NOLINT
     }
     return Status::OK();
 }
 
 Status MathFunctions::rand_close(FunctionContext* context, FunctionContext::FunctionStateScope scope) {
+    if (scope == FunctionContext::THREAD_LOCAL) {
+        auto state = reinterpret_cast<std::mt19937_64*>(context->get_function_state(FunctionContext::THREAD_LOCAL));
+        delete state;
+    }
     return Status::OK();
 }
 
 StatusOr<ColumnPtr> MathFunctions::rand(FunctionContext* context, const Columns& columns) {
-    int32_t num_rows = ColumnHelper::get_const_value<TYPE_INT>(columns[columns.size() - 1]);
-    void* state = context->get_function_state(FunctionContext::THREAD_LOCAL);
+    std::mt19937_64* generator =
```

Comment:
declare static thread local here

Reply:
Thanks, I modified it based on your suggestions, and the test performance is in line with expectations. Please help to review it again. @stdpain